packages/util-package-renderer/lib/js/js-helper.js

```diff
@@ -30,9 +35,21 @@ const transpileJS = async (packageRoot, demoCodeFolder) => {
 	}
 
 	// Create a rollup bundle
+	// - Handle import and require
+	// - Transpile using babel
 	try {
 		bundle = await rollup.rollup({
 			input: path.join(packageRoot, demoCodeFolder, 'main.js'),
+			plugins: [commonjs({
+				sourcemap: false
+			}),
+			nodeResolve(),
+			...minify ? [terser()] : [],
```

Comment:
This line hurts my poor head.  Spread syntax on a bool? 🤔

Reply:
yeah, it's confusing, but seemed the neatest way to conditionally add the `terser()` command into the array.
You only run terser if you are minifying, and it has to appear at that point in the array, so that it runs after nodeResolve and before babel. So the spread basically says merge are new array into this exact point in the array - so if minify is true terser gets added, otherwise and empty array i.e. nothing

The only way I can think is to have two entirely different configs for minify/not

Reply:
I get what it is doing thank you, but I'm pretty sure you can't use spread syntax on a boolean, which is what this code looks like it's doing. So you are saying the spread runs on what the ternary evaluates-to, not the bool? Can we rewrite this to be a bit clearer please?

Also, the indentation now needs fixing on these plugin values.

Reply:
Yep, exactly that, it runs on the result so either `[]` or `[terser()]`. I'll see if I can make it clearer but not sure how, there is only so much you can do inside an array

Reply:
I've added brackets around the ternary to hopefully make it clearer that the spread happens on the result, and also added a comment line